app-service-template/res/configuration.toml

```
@@ -105,11 +112,20 @@ TokenFile = "/tmp/edgex/secrets/new-app-service/secrets-token.json"
   Port = 59881
 
   [Clients.core-command]
-  Protocol = "http"
-  Host = "localhost"
-  Port = 59882
-
-  [Clients.support-notifications]
+#  Protocol = "http"
+#  Host = "localhost"
+#  Port = 59882
+  # Choose to use new messaging based communication for Command Client.
+  # Only valid if MessageBus Trigger is used
+  # If prefer REST, remove below and uncomment above
+  UseMessageBus = true
+  [Clients.core-command.Topics]
+  QueryRequestTopic = "edgex/core/commandquery/request"
+  QueryResponseTopic = "edgex/core/commandquery/response/#"
```

```suggestion
  QueryResponseTopic = "edgex/core/commandquery/response"
```

This should fix the timeout error

---

Ok, that resolved Query Response. But now getting timeout on the command response. I tried remove `/#` from `CommandResponseTopic = "edgex/core/command/response/#"` but it didn't help.

---

hmmm the configuration looks good to me and it works correctly on my side.
Did you see the similar logs in the core-command ?
```
level=DEBUG ts=2022-10-21T02:06:09.908693012Z app=core-command source=internal.go:114 msg="Command request received on internal MessageBus. Topic: edgex/core/command/request/Random-Boolean-Device/Bool/get, Correlation-id: e9284904-d188-47cb-8681-35b0eb0db9e8 "
level=DEBUG ts=2022-10-21T02:06:09.910602655Z app=core-command source=internal.go:154 msg="Command request sent to internal MessageBus. Topic: edgex/device/command/request/device-virtual/Random-Boolean-Device/Bool/get, Correlation-id: e9284904-d188-47cb-8681-35b0eb0db9e8"
level=DEBUG ts=2022-10-21T02:06:09.91118834Z app=core-command source=internal.go:55 msg="Command response received on internal MessageBus. Topic: edgex/device/command/response/device-virtual/Random-Boolean-Device/Bool/get, Correlation-id: e9284904-d188-47cb-8681-35b0eb0db9e8 "
level=DEBUG ts=2022-10-21T02:06:09.911449236Z app=core-command source=internal.go:75 msg="Published response message to internal MessageBus on topic 'edgex/core/command/response/Random-Boolean-Device/Bool/get'"
```

---

Ohh, This is probably because Device Virtual still needs latest SDK in order to receive and respond to the command request.

---

I'm using the latest device-virtual image and seeing no issue.

---

I built Device Virtual locally with latest SDK and it works now.